src/screens/Register.jsx

```diff
@@ -21,21 +32,53 @@ const Register = () => {
   };
 
   const assignUserRole = async (email, values) => {
-    const account = await assignRole({ email, ...values });
-    console.log(account);
+    if (role === "candidate") {
+      const account = await assignCandidate({ email, ...values });
+      console.log(account);
+    } else {
+      const account = await assignCompany({ email, ...values });
+      console.log(account);
+    }
   };
 
-  if (isLoading) return <Loader />;
+  if (isLoading && role) return <Loader />;
 
   return (
     <>
       <PageTitle>
         <h3>Register</h3>
       </PageTitle>
-      <CandidateForm
-        onSubmit={onFormSubmit}
-        fields={CREATE_CANDIDATE_FIELDS}
-      ></CandidateForm>
+      {!role ? (
+        <div style={styleScreen}>
+          <PageTitle>
+            <h4>Are you a candidate or a company?</h4>
+          </PageTitle>
+          <Button
+            variant="secondary"
+            size="huge"
+            action={() => setRole("candidate")}
+          >
+            Candidate
+          </Button>
+          <Button
+            variant="secondary"
+            size="huge"
+            action={() => setRole("company")}
+          >
+            Company
+          </Button>
+        </div>
+      ) : role === "candidate" ? (
+        <CandidateForm
```

Comment:
Instead of duplicating the `<CandidateForm/>` component, you could just change its `fields` prop.

`fields={role === "candidate" ? CREATE_CANDIDATE_FIELDS : CREATE_COMPANY_FIELDS }`